quickwit/quickwit-control-plane/src/ingest/ingest_controller.rs

```diff
@@ -191,6 +208,25 @@ impl IngestController {
         }
     }
 
+    pub(crate) async fn handle_local_shards_update(
+        &mut self,
+        local_shards_update: LocalShardsUpdate,
+        model: &mut ControlPlaneModel,
+        progress: &Progress,
+    ) {
+        let shard_stats = model.update_shards(
+            &local_shards_update.source_uid,
+            &local_shards_update.shard_infos,
+        );
+        if shard_stats.avg_ingestion_rate >= SCALE_OUT_SHARDS_THRESHOLD_MIB_PER_SEC {
+            self.try_scale_out_shards(local_shards_update.source_uid, shard_stats, model, progress)
```

Comment:
i think there is an important corner case if you have only one shard.

Reply:
It's handled in `try_scale_out_shards`, but I'll move the check here.